tests/testsuite/feature_unification.rs

```diff
@@ -203,3 +1005,362 @@ fn unstable_config_on_stable() {
+            r#"
+                [workspace]
+                members = ["bar"]
+
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2018"
+
+                [dependencies]
+                common = "1.0"
+                bar = "1.0"
+
+                [build-dependencies]
+                common = { version = "1.0", features = ["opt_dep"] }
+
+                [dev-dependencies]
+                common = { version="1.0", features=["dev-feat"] }
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.1.0"
+                edition = "2018"
+            "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    p.cargo("fix --edition --allow-no-vcs --workspace")
+        .env("CARGO_RESOLVER_FEATURE_UNIFICATION", "selected")
+        .arg("-Zfeature-unification")
+        .masquerade_as_nightly_cargo(&["feature-unification"])
+        .with_status(0)
+        .with_stderr_data(
+            str![[r#"
+[MIGRATING] Cargo.toml from 2018 edition to 2021
+[UPDATING] `dummy-registry` index
+[LOCKING] 3 packages to latest compatible versions
+[DOWNLOADING] crates ...
+[DOWNLOADED] common v1.0.0 (registry `dummy-registry`)
+[DOWNLOADED] bar v1.0.0 (registry `dummy-registry`)
+[DOWNLOADED] opt_dep v1.0.0 (registry `dummy-registry`)
+[MIGRATING] bar/Cargo.toml from 2018 edition to 2021
+[NOTE] Switching to Edition 2021 will enable the use of the version 2 feature resolver in Cargo.
+This may cause some dependencies to be built with fewer features enabled than previously.
+More information about the resolver changes may be found at https://doc.rust-lang.org/nightly/edition-guide/rust-2021/default-cargo-resolver.html
+When building the following dependencies, the given features will no longer be used:
+
+  common v1.0.0 removed features: dev-feat, f1, opt_dep
+  common v1.0.0 (as host dependency) removed features: dev-feat, f1
+
+The following differences only apply when building with dev-dependencies:
+
+  common v1.0.0 removed features: f1, opt_dep
+
+[CHECKING] opt_dep v1.0.0
+[CHECKING] bar v1.0.0
+[CHECKING] bar v0.1.0 ([ROOT]/foo/bar)
+[MIGRATING] bar/src/lib.rs from 2018 edition to 2021
+[CHECKING] common v1.0.0
+[CHECKING] foo v0.1.0 ([ROOT]/foo)
+[MIGRATING] src/lib.rs from 2018 edition to 2021
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]]
+            .unordered(),
+        )
+        .run();
+}
+
+#[cargo_test]
+fn package_feature_unification_grandchild_cli_features() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [workspace]
+                resolver = "2"
+                members = ["parent", "child", "grandchild"]
+            "#,
+        )
+        .file(
+            "grandchild/Cargo.toml",
```

Comment:
Whats the intention of this test?

Reply:
I've renamed the second one to `feature_unification_of_cli_features_within_workspace`. I didn't come up with anything more descriptive yet reasonable than `package_feature_unification_cli_features` for the first one so I've left that as is.